httputil/json.go

```diff
@@ -0,0 +1,31 @@
+package httputil
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+)
+
+const (
+	JSONContentType = "application/json"
+
+	StatusError = "error"
+	StatusOk    = "ok"
+)
+
+func WriteJSONResponse(w http.ResponseWriter, code int, resp interface{}) {
+	enc, err := json.Marshal(resp)
+	if err != nil {
+		log.Printf("Failed JSON-encoding HTTP response: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+
+	_, err = w.Write(enc)
+	if err != nil {
+		log.Printf("Failed sending HTTP response body: %v", err)
```

Comment:
We have the same log problem here. Maybe instead this func should return an error and let the caller decide how to handle it?